spec/controllers/votes_controller_spec.rb

```diff
@@ -1,4 +1,83 @@
 require 'rails_helper'
 
 RSpec.describe VotesController, type: :controller do
+    let(:user1) { create(:user_with_voteable_status) }
+    let(:user2) { create(:user_with_voteable_status) }
+    let(:user3) { create(:user_without_voteable_status) }
+    before do
+      create_list(:question, 5, user: user1)
+      create_list(:answer, 5, user: user2)
+    end
+
+  context "when user votes another user's resource" do
+    before do
+      request.headers['Authorization'] = "Token token=#{user2.refresh_token}"
+    end
+
+    it "upvotes another user's resource" do
+      post :upvote, {resource_name: 'questions', resource_id: 1}
+      expect(response.body).to eq "{\"response\":1}"
+      expect(response.status).to eq 200
+    end
```

Comment:
If you don't mind leveraging on already existing helpers. You could check out the `spec/support` dir. There a number of helpers already provided. E.g `ValidRequest` for valid token and `parsed_json` for response. So, here we may not need to do  `(response.body).to eq "{\"response\":1}".